examples/jsm/nodes/utils/ArrayElementNode.js

```diff
@@ -19,13 +20,75 @@ class ArrayElementNode extends Node { // @TODO: If extending from TempNode it br
 
 	}
 
-	generate( builder ) {
+	setup( builder ) {
+
+		if ( this.node.isStorageBufferNode && ! builder.isAvailable( 'storageBuffer' ) ) {
+
+			if ( ! this.node.instanceIndex ) {
+
+				const attribute = this.node.value;
+
+				if ( attribute.pbo === undefined ) {
+
+					const square = Math.sqrt( attribute.array.length / 4 );
+					const width = Math.floor( square );
+					const height = Math.ceil( square );
+
+					const pboTexture = new DataTexture( attribute.array, width, height, RGBAFormat, FloatType );
```

Comment:
The circular-reference problem is because of  `UniformNode` in `ArrayElementNode`, `UniformNode` will use `ShaderNode` and `ShaderNode` will include try `ArrayElementNode` again.

I think it would be better to move this uniform creation block to `GLSLNodeBuilder` something like `builder.getPBOUniform()` for example, this could fix the circular reference problem too.

Reply:
I think we have to consider a new Node, similar to what was done in `BufferAttributeNode` and `buffer.toAttribute()` as usage. It could be `buffer.toTexture()`

Reply:
I agree, this feels like something that will better be placed in GLSLNodeBuilder.

Reply:
Also consider move it to `StorageBufferNode.element()` to avoid checks a little, and return and new Node to keep operations more organized, it can even be a `StorageArrayElementNode`.

Reply:
Thanks for the feedbacks!

The downside of `buffer.toTexture()` would be that this implies that somehow the developer would expect that something happens in the `WebGPU` Backend too.
This feature is more like a patch to fix the fact WebGL cannot access other element of array buffers rather than a new feature such as `toAttribute()`.

I think I can try to move all this part to `GLSLNodeBuilder` but I'm not sure on how to create a Node here. Maybe once everything works we can think about creating a new Node specific to `Pixel Buffer Object`, but once again it's only a WebGL thing.

Reply:
I'll be happy to help you with this, let me know when the PR is ready for review so I can analyze it again.

About the example what do you think about renaming it to `webgpu_storage_sorting`?

Reply:
Sounds good to me! Thanks a lot for this 😊